### ISSUE TITLE
EditButton inside <Tab> to be "tab-aware"

### ISSUE THREAD

**Issue by tomwolfgang** [marmelab/react-admin#2742]
**Is your feature request related to a problem? Please describe.**
When in a **Show** form that has a **TabbedShowLayout** - clicking the Edit button (of the default actions implementation) will switch to the **Edit** form but jump to the first Tab in the form

**Describe the solution you'd like**
Clicking on Edit will remember the tab you started from and select it in the edit form (if it exists)

**Describe alternatives you've considered**
Implement a custom EditButton that takes the tab name from the url (.../show/tabname) and then links to it.

**Additional context**
What if the Edit form doesn't have this tab - needs to be robust.

Thanks!

**Comment by djhi** [comment]
I think you missed some words here:
> When in a form that has a - clicking the Edit button (of the default actions implementation) will switch to the form but jump to the first Tab in the form

Can you rephrase it ?

**Comment by tomwolfgang** [comment]
I'll try :)
Say I have an entity: Car
The Car has a **Show** form:
  <Show ...
    <TabbedShowLayout ...
      <Tab label="Engine" ...
       <Tab label="Model" ...

The Car also has an **Edit** form:
  <Edit ...
    <TabbedForm...
      <FormTab label="Engine" ...
       <FormTab label="Model" ...


Say the user is in the <Show> form/view, selects the "Model" Tab and then decides he wants to edit the Model information - so he clicks the Edit button **and expects to** be in the <Edit> form/view with the "Model" FormTab selected.
Right now, clicking <Edit> will show him the <Edit> form but the "Engine" FormTab is selected.

Hope this makes more sense,
thanks

**Comment by djhi** [comment]
I see. As you mentionned, it assumes the Show and Edit views have the same layout. We won't implement that in the core but you should be able to achieve it with a custom `Edit` button.
